src/app/pages/test-report-dashboard/test-report-dashboard.page.ts

```diff
@@ -181,7 +181,9 @@ export class TestReportDashboardPage extends TestReportBasePageComponent impleme
       case ModalEvent.TERMINATE:
         this.store$.dispatch(GradeChanged(null));
         this.store$.dispatch(TerminateTestFromTestReport());
-        await this.router.navigate([TestFlowPageNames.DEBRIEF_PAGE]);
+        if (this.adi3AssessmentProvider.isTestReportPopulated(this.testDataADI3)) {
+          await this.router.navigate([TestFlowPageNames.DEBRIEF_PAGE]);
+        } else await this.router.navigate([TestFlowPageNames.NON_PASS_FINALISATION_PAGE]);
```

Comment:
Think we should avoid if possible doing if/else blocks inside a switch (just for readability purposes).

```
const populatedTestReport = this.adi3AssessmentProvider.isTestReportPopulated(this.testDataADI3);
await this.router.navigate(populatedTestReport ? [TestFlowPageNames.DEBRIEF_PAGE] : [TestFlowPageNames.NON_PASS_FINALISATION_PAGE]); 
```